airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py

```diff
@@ -0,0 +1,115 @@
+from pipelines import main_logger
+from dataclasses import dataclass
+from pathlib import Path
+from pipelines.actions.environments import with_installed_pipx_package, with_python_base
+from pipelines.bases import Step, StepResult
+from pipelines.contexts import PipelineContext
+
+import dagger
+
+from typing import List, Optional
+
+
+@dataclass(frozen=True)
+class MountPath:
+    path: Path
+    optional: bool = False
+
+    def __post_init__(self):
+        if not self.path.exists():
+            message = f"{self.path} does not exist."
+            if self.optional:
+                raise FileNotFoundError(message)
+            else:
+                main_logger.warning(message)
+
+    def __str__(self):
+        return str(self.path)
+
+    @property
+    def is_file(self) -> bool:
+        return self.path.is_file()
+
+
+class SimpleCIStep(Step):
+    def __init__(
+        self,
+        title: str,
+        context: PipelineContext,
+        paths_to_mount: List[MountPath] = [],
+        internal_tools: List[MountPath] = [],
+        secrets: dict[str, dagger.Secret] = {},
+        env_variables: dict[str, str] = {},
+        working_directory: str = "/",
+        command: Optional[List[str]] = None,
+    ):
+        """A simple step that runs a given command in a container.
```

Comment:
Suggestion: what about naming this `DockerRunStep`? Similar to `PoetryRunStep`, but I feel like it gives an indication of what will happen with the arguments

Reply:
Although that might be a bit confusing since you don't pass in any info about what container to run (since it comes in through the context, right?)

Reply:
Ah I like this! Should I keep `Simple`?

Im leaning towards yes, because Im not sure this should be the primary abstraction for all use cases. Just basic ones

Reply:
> Im not sure this should be the primary abstraction for all use cases. Just basic ones

Makes sense to me - encourage people to build their own if things get complicated vs hacking more stuff into it. `SimpleDockerRunStep` seems good 👍🏻 Maybe a note in the docstring about "meant to serve as a base to handle simple use cases"